GarbageCan.Application/XP/Services/XpCalculatorService.cs

```diff
@@ -1,24 +1,37 @@
 ﻿using System;
+using GarbageCan.Application.Boosters.Commands;
 using GarbageCan.Application.Common.Interfaces;
+using MediatR;
 
 namespace GarbageCan.Application.XP.Services
 {
     public class XpCalculatorService : IXpCalculatorService
     {
-        public double XpEarned(string message)
+        private readonly IMediator _mediator;
+
+        public XpCalculatorService(IMediator mediator)
         {
+            _mediator = mediator;
+        }
+
+        public double XpEarned(string message, ulong guildId)
+        {
+            float GetMultiplier() =>
```

Comment:
Doesn't look like there's really a need for a local method here. Just more code to read. The mediator call can be just used down on line 29 instead.